lib/config/flat-config-schema.js

```diff
@@ -0,0 +1,362 @@
+/**
+ * @fileoverview Flat config schema
+ * @author Nicholas C. Zakas
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @typedef ObjectPropertySchema
+ * @property {Function|string} merge The function or name of the function to call
+ *      to merge multiple objects with this property.
+ * @property {Function|string} validate The function or name of the function to call
+ *      to validate the value of this property.
+ */
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+const ruleSeverities = new Map([
+    [0, 0], ["off", 0],
+    [1, 1], ["warn", 1],
+    [2, 2], ["error", 2]
+]);
+
+const globalVariablesValues = new Set([
+    true, "writable", "writeable",
+    false, "readonly",
+    "off"
+]);
```

Comment:
Aside from these, we also allow: `"true"` (string), `null`, `"false"` (string), and `"readable"`.

https://github.com/eslint/eslintrc/blob/62ea4bd74b78fbeff12ffb21f1f978817601d4d1/lib/shared/config-ops.js#L108-L129